### DISCUSSION TITLE
StreamReader ReadFragment

### DISCUSSION THREAD

**Discussion by Yedidyah27** [microsoft/OpenAPI.NET#1096] [Q&A] [answered]
We have recently upgraded OpenApi version from 1.2.3 to 1.4.4. We were using `OpenApiStreamReader` `ReadFragment `method to deserialize a json to `OpenApiOperation `class. This is no longer supported. I noticed though that `OpenApiStringReader` hasnt changed and still supports this. Any reason why these two readers are different? Why was `ReadFragment `changed to only support `IOpenApiReferencable`?

**Comment by baywet** [comment, marked as answer]
Hi @yediyah27

Thank you for using the SDK and for reaching out.

I believe this was a regression which should have been opened as an issue at the time for investigation. Is this an issue you're still facing with versions 2.X or 3.X of the library?

Let us know if you have any additional comments or questions.
